File: models/auditLog.js
class AuditLog {
    constructor(id, timestamp, actorId, actionType, targetResource, targetResourceId, ipAddress, userAgent, successStatus, details) {
        this.id = id;
        this.timestamp = timestamp;
        this.actorId = actorId;
        this.actionType = actionType;
        this.targetResource = targetResource;
        this.targetResourceId = targetResourceId;
        this.ipAddress = ipAddress;
        this.userAgent = userAgent;
        this.successStatus = successStatus;
        this.details = details;
    }
}

module.exports = AuditLog;

File: Sources/AuditLog.swift
import Foundation

struct AuditLog: Identifiable {
    // Represents the flexible structure of a JSONB field for audit details.
    struct AuditDetails: Codable {
        let parameters: [String: String]?
        let beforeValue: String?
        let afterValue: String?
    }

    let id: UUID
    let timestamp: Date
    let actorId: UUID?
    let actionType: String
    let targetResource: String?
    let targetResourceId: UUID?
    let ipAddress: String?
    let userAgent: String?
    let successStatus: Bool
    let details: AuditDetails?
}
